galleries/examples/event_handling/viewlims.py

```diff
@@ -77,17 +90,15 @@ def ax_update(self, ax):
 ax2.imshow(Z, origin='lower',
            extent=(md.x.min(), md.x.max(), md.y.min(), md.y.max()))
 
-rect = UpdatingRect(
+rect = Rectangle(
     [0, 0], 0, 0, facecolor='none', edgecolor='black', linewidth=1.0)
-rect.set_bounds(*ax2.viewLim.bounds)
 ax1.add_patch(rect)
```

Comment:
There's no need for this patch now; you can make the `RectangleSelector` interactive and it will stay visible.

Reply:
This patch is on ax1 (left plot) to draws the `Rectangle` around the extent selected by the `RectangleSelector` in ax2 (right plot) so it seems necessary.

Reply:
Ah, I missed that it was on `ax2`, but it should be on `ax1` to replace the `Rectangle` entirely.

Reply:
I see what you meant - I think the original idea of this example is to demonstrate the fractal structure of the [Mandelbrot set](https://en.wikipedia.org/wiki/Mandelbrot_set). The user can keep zooming in on the selected area on the right subplot and continue to see more recursive details, while keeping track of this area's location by looking at the left subplot (like a map). From a code design perspective I agree it's cleaner to have one single `RectangleSelector` on `ax1`. Let me know what you think is a better option.

Reply:
Yes, my suggestion in the issue was to replace the `Rectangle` by a `RectangleSelector`, with suitable settings to keep it visible.